### PR TITLE
Move Corporate Information Index page to the GOV.UK Design System

### PR DESCRIPTION
**Pull request by davidgisbey**
## Description 

This PR moves the Organisation Corporate information index page & Worldwide Organisation Corporate information index page to the to the GOV.UK Design System. 

The reason for this is that it is a shared view and all lives in one controller action.

This is slightly more complex than at first glance looking at the page for 2 reasons:

1. We needed to add pagination which wasn't immediately apparent as it's 100 items per page
2. We had to add secondary navigation tabs with the more tabs broken out into additional tabs

This work is hidden behind the `Preview design system permission`

##### Before

image

##### After

image

#### With pagination 

The current per page for this is 100, but i've changed it to 5 here just for demonstration

##### Before

image

##### After

image

##### Before

image

##### After

image

#### With pagination 

The current per page for this is 100, but i've changed it to 5 here just for demonstration

##### Before

image

##### After

image

## Trello card 

https://trello.com/c/p6vopzC9/22-org-corporate-information-index-page
https://trello.com/c/qL6X06aL/44-worldwide-org-corporate-information-page

Co-authored-by: Conor Lowney <conor.lowney@digital.cabinet-office.gov.uk>

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

### CODE REVIEW

File: app/assets/stylesheets/admin/views/_corporate-information-index.scss [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    cursor: not-allowed;
+  }
+
+  a {
+    color: $govuk-link-colour;
+  }
+
+  a:hover {
+    color: $govuk-link-hover-colour;
+  }
```
Comment (davidgisbey) [inline app/assets/stylesheets/admin/views/_corporate-information-index.scss:41]: I imagine there's a far better way to import govuk-link behaviour than this. The ways i found complained in the linter though. V open to suggestions!

File: app/assets/stylesheets/admin/views/_corporate-information-index.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+.govuk-tag--s {
```
[the remark this answers is not in the document]
Comment (davidgisbey) [reply app/assets/stylesheets/admin/views/_corporate-information-index.scss:1]: yea good shout. I don't know if we have a sort of generalised file for extending/overriding classes. wdyt?

[the remark this answers is not in the document]
Comment (ollietreend) [reply app/assets/stylesheets/admin/views/_corporate-information-index.scss:1]: Yeah – I think we should have a 'proper' place to put these sorts of things. What we're trying to do is extend or change a GOV.UK Design System component within our application.

Content Publisher has an "overwrites" file for these situations:
[app/assets/stylesheets/utilities/_overwrites.scss](https://github.com/alphagov/content-publisher/blob/main/app/assets/stylesheets/utilities/_overwrites.scss)

How about we introduce the same?

So this `.govuk-tag--s` rule would live in the file:
```
app/assets/stylesheets/admin/_overrides.scss
```

(I personally think "overrides" is a more suitable name than "overwrites" - but I don't hold that opinion strongly)

I expect we won't need to do this all that frequently, so let's just go with a single file for now. If we find we're overriding/extending component rules all the time, then we might come back and change it into a directory.

[the remark this answers is not in the document]
Comment (davidgisbey) [reply app/assets/stylesheets/admin/views/_corporate-information-index.scss]: Amazing. Have actioned. Thanks!

**Comment by davidgisbey** [comment]
Have sorted the stylesheet. Thanks!

**Review by ollietreend — APPROVED** [review]
Looks good to me 👍🏻 

Thanks for humouring me on the pagination stuff! 🙈

**Comment by davidgisbey** [comment]
All good i think you're probs right tbh.

**Comment by cbernardojr-gds** [comment]
@davidgisbey Few changes:

- Pagination: remove the top pagination, and keep the bottom one
- Pagination: The style seems different from the GOV.UK pagination (see image)
- Call to action: "Create new corporate information page" in green
- Edit as a link on the right column
- Optional: Merge the states and warnings in the same column and use tags for states (see image)
- Optional: Combine the author and updated time info  (see image)

screencapture-gds-wh-herokuapp-v2-organisation-pages-2023-03-15-14_24_18

Reference:
https://gds-wh.herokuapp.com/v2/organisation/pages
Password: wh1t3h@11

**Pull request merged**